Migrate user data from OpenList UID to tokens

kasperg opened the pull request:
This PR adds migration capabilities to the Ding React module.

When a user logs in their legacy OpenList UID is retrieved and used to migrate to their new token-based ID.

This is handled serverside and requires some new configuration options which are added accordingly.

Please check individual commit messages for more comments.

kasperg committed:
Add configuration of React module

This allows administrators to configure which instances of the
services to use.

kasperg committed:
Do not drop the openlist userid_field

We need it when performing migrations.

kasperg committed:
Migrate users on login

kasperg committed:
Add timeout handling for user list migration

We want a low timeout to prevent the system from stalling during
login - especially if the list services perform poorly. This effect
will multiply if many users wait for a long time for migration to
complete. We prefer a lower timeout which in the case of slowness
will migrate a user the next time they log in.

5 seconds seems like a suitable number. We do not expect the
migration to take more that 1 second but in the case of request
pileup we do not want them to wait any longer than that.

kasperg committed:
Move handling of openlist_uid field to Ding React module

It was previously handled by Ting Openlist but that module will be
disabled. We still need the OpenList UID for the migration process
so we move the handling to the new module.

kasperg committed:
Refactored user list data migration from hook to function

This makes the hook implementation simpler and may make it easier to
invoke a migration under different circumstances.

xendk commented on modules/p2/ting_openlist/ting_openlist.install:
```diff
@@ -79,7 +79,8 @@ function ting_openlist_uninstall() {
 
   // This would be the proper thing to do when uninstalling, but we need the uid
-  // to migrate lists to the new system, so leave it in place.
+  // to migrate lists to the new system, so the field is not handled by
```
"not handled"?

xendk reviewed (commented):
Looks good, but the build fails.

kasperg committed:
Do not try to perform migrations if we cannot get a token

Without this we cannot perform login on installations without
either Connie or Adgangsplatformen enabled.
